### PR TITLE
[6585] Support array data in /api/1/profile/set

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6585

### CODE REVIEW

File: ui/app/src/services/configuration.ts [hunk 1]
```diff
@@ -137,15 +137,6 @@ export function deserializeActiveTargetingModelData<T extends Object>(
   data: T,
   contentTypeFields: LookupTable<ContentTypeField>
 ): ContentInstance {
-  Object.keys(data).forEach((modelKey) => {
```
Comment (jvega190) [inline ui/app/src/services/configuration.ts:140]: This is not needed anymore as the service now returns the checkbox-group data as an array, not a comma-separated string

File: ui/app/src/state/epics/audiencesPanel.ts [hunk 2]
```diff
@@ -25,7 +25,8 @@ import {
   SET_ACTIVE_TARGETING_MODEL,
   SET_ACTIVE_TARGETING_MODEL_COMPLETE,
   setActiveTargetingModelComplete as setActiveTargetingModelCompleteAction,
-  setActiveTargetingModelFailed
+  setActiveTargetingModelFailed,
+  setActiveTargetingModelFailed as setActiveTargetingModelFailedAction
```
Comment (rart) [inline ui/app/src/state/epics/audiencesPanel.ts]: Why import twice with different names?

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — COMMENTED** [review]
Why are there so many differences between this PR and the one to [4.1.x](https://github.com/craftercms/studio-ui/pull/3848/files)?

**Comment by jvega190** [comment]
> Why are there so many differences between this PR and the one to [4.1.x](https://github.com/craftercms/studio-ui/pull/3848/files)?

@rart there was this PR already merged for 4.1.x https://github.com/craftercms/studio-ui/pull/3734

File: ui/app/src/state/epics/audiencesPanel.ts (hunk 2 above)
Comment (jvega190) [reply ui/app/src/state/epics/audiencesPanel.ts]: This was a mistake, removed the duplicate

**Review by rart — APPROVED** [review]

**Pull request merged**